src/azure-cli-core/azure/cli/core/_util.py

```diff
@@ -91,7 +91,7 @@ def to_camelcase(s):
         return re.sub(KEYS_CAMELCASE_PATTERN, lambda x: x.group(1).upper(), s)
 
     if isinstance(obj, dict):
-        return {k: todict(v) for (k, v) in obj.items()}
+        return {to_camelcase(k): todict(v) for (k, v) in obj.items()}
```

Comment:
`to_camelcase()` is used for the `hasattr(obj, '__dict__')` if statement below so looks like it was missed here.